### PR TITLE
[cmake] Add minor version to library SONAME

### LINKED ISSUE #76273: Change library SONAMEs to include minor version
As suggested in #75743 and https://discourse.llvm.org/t/rfc-name-the-first-release-from-a-branch-n-1-0-instead-of-n-0-0/75384 there is a desire to switch from just the major version (e.g. "18") to instead use major+minor (e.g. "18.0" or "18.1").

### PR DESCRIPTION
**Pull request by tstellar**
We need to do this now that we are bumping the minor release number when we create the release branch.

This also results in a slight change to the library names for LLVM.  The main library now has a more convential library name: 'libLLVM.so.$major.$minor'.  The old library name: libLLVM-$major.so is now a symlink that points to the new library.  However, the symlink is not present in the build directory.  It is only present in the install directory.

The library name was changed because it helped to keep the CMake changes more simple.

Fixes #76273

### CODE REVIEW

File: llvm/cmake/modules/AddLLVM.cmake [hunk 1]
```diff
@@ -647,8 +647,8 @@ function(llvm_add_library name)
       set_target_properties(${name}
         PROPERTIES
         # Since 4.0.0, the ABI version is indicated by the major version
```
Comment (jyknight) [inline llvm/cmake/modules/AddLLVM.cmake:649]: Comment needs updating.

**Review by jyknight — APPROVED** [review]
I'm not an expert in this area, but this feels plausible to me.

**Comment by kraj** [comment]
its creating symlink `libLLVM-<major-ver>.so` pointing to `libLLVM.so` I think it would be nicer for packagers if it was pointing to newly renamed `libLLVM.so.maj.minor`

**Pull request merged**